// @pulumi/aws/opsworks/application.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an OpsWorks application resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 * import * from "fs";
 *
 * const foo_app = new aws.opsworks.Application("foo-app", {
 *     shortName: "foobar",
 *     stackId: aws_opsworks_stack.main.id,
 *     type: "rails",
 *     description: "This is a Rails application",
 *     domains: [
 *         "example.com",
 *         "sub.example.com",
 *     ],
 *     environments: [{
 *         key: "key",
 *         value: "value",
 *         secure: false,
 *     }],
 *     appSources: [{
 *         type: "git",
 *         revision: "master",
 *         url: "https://github.com/example.git",
 *     }],
 *     enableSsl: true,
 *     sslConfigurations: [{
 *         privateKey: fs.readFileSync("./foobar.key"),
 *         certificate: fs.readFileSync("./foobar.crt"),
 *     }],
 *     documentRoot: "public",
 *     autoBundleOnDeploy: "true",
 *     railsEnv: "staging",
 * });
 * ```
 *
 * ## Import
 *
 * Opsworks Application can be imported using the `id`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:opsworks/application:Application test <id>
 * ```
 */
class Application extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["appSources"] = state ? state.appSources : undefined;
            resourceInputs["autoBundleOnDeploy"] = state ? state.autoBundleOnDeploy : undefined;
            resourceInputs["awsFlowRubySettings"] = state ? state.awsFlowRubySettings : undefined;
            resourceInputs["dataSourceArn"] = state ? state.dataSourceArn : undefined;
            resourceInputs["dataSourceDatabaseName"] = state ? state.dataSourceDatabaseName : undefined;
            resourceInputs["dataSourceType"] = state ? state.dataSourceType : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["documentRoot"] = state ? state.documentRoot : undefined;
            resourceInputs["domains"] = state ? state.domains : undefined;
            resourceInputs["enableSsl"] = state ? state.enableSsl : undefined;
            resourceInputs["environments"] = state ? state.environments : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["railsEnv"] = state ? state.railsEnv : undefined;
            resourceInputs["shortName"] = state ? state.shortName : undefined;
            resourceInputs["sslConfigurations"] = state ? state.sslConfigurations : undefined;
            resourceInputs["stackId"] = state ? state.stackId : undefined;
            resourceInputs["type"] = state ? state.type : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.stackId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'stackId'");
            }
            if ((!args || args.type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["appSources"] = args ? args.appSources : undefined;
            resourceInputs["autoBundleOnDeploy"] = args ? args.autoBundleOnDeploy : undefined;
            resourceInputs["awsFlowRubySettings"] = args ? args.awsFlowRubySettings : undefined;
            resourceInputs["dataSourceArn"] = args ? args.dataSourceArn : undefined;
            resourceInputs["dataSourceDatabaseName"] = args ? args.dataSourceDatabaseName : undefined;
            resourceInputs["dataSourceType"] = args ? args.dataSourceType : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["documentRoot"] = args ? args.documentRoot : undefined;
            resourceInputs["domains"] = args ? args.domains : undefined;
            resourceInputs["enableSsl"] = args ? args.enableSsl : undefined;
            resourceInputs["environments"] = args ? args.environments : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["railsEnv"] = args ? args.railsEnv : undefined;
            resourceInputs["shortName"] = args ? args.shortName : undefined;
            resourceInputs["sslConfigurations"] = args ? args.sslConfigurations : undefined;
            resourceInputs["stackId"] = args ? args.stackId : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Application.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Application resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Application(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Application.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Application.__pulumiType;
    }
}
exports.Application = Application;
/** @internal */
Application.__pulumiType = 'aws:opsworks/application:Application';
