// @pulumi/aws/opsworks/memcachedLayer.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MemcachedLayer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an OpsWorks memcached layer resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const cache = new aws.opsworks.MemcachedLayer("cache", {stackId: aws_opsworks_stack.main.id});
 * ```
 */
class MemcachedLayer extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["allocatedMemory"] = state ? state.allocatedMemory : undefined;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["autoAssignElasticIps"] = state ? state.autoAssignElasticIps : undefined;
            resourceInputs["autoAssignPublicIps"] = state ? state.autoAssignPublicIps : undefined;
            resourceInputs["autoHealing"] = state ? state.autoHealing : undefined;
            resourceInputs["cloudwatchConfiguration"] = state ? state.cloudwatchConfiguration : undefined;
            resourceInputs["customConfigureRecipes"] = state ? state.customConfigureRecipes : undefined;
            resourceInputs["customDeployRecipes"] = state ? state.customDeployRecipes : undefined;
            resourceInputs["customInstanceProfileArn"] = state ? state.customInstanceProfileArn : undefined;
            resourceInputs["customJson"] = state ? state.customJson : undefined;
            resourceInputs["customSecurityGroupIds"] = state ? state.customSecurityGroupIds : undefined;
            resourceInputs["customSetupRecipes"] = state ? state.customSetupRecipes : undefined;
            resourceInputs["customShutdownRecipes"] = state ? state.customShutdownRecipes : undefined;
            resourceInputs["customUndeployRecipes"] = state ? state.customUndeployRecipes : undefined;
            resourceInputs["drainElbOnShutdown"] = state ? state.drainElbOnShutdown : undefined;
            resourceInputs["ebsVolumes"] = state ? state.ebsVolumes : undefined;
            resourceInputs["elasticLoadBalancer"] = state ? state.elasticLoadBalancer : undefined;
            resourceInputs["installUpdatesOnBoot"] = state ? state.installUpdatesOnBoot : undefined;
            resourceInputs["instanceShutdownTimeout"] = state ? state.instanceShutdownTimeout : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["stackId"] = state ? state.stackId : undefined;
            resourceInputs["systemPackages"] = state ? state.systemPackages : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
            resourceInputs["useEbsOptimizedInstances"] = state ? state.useEbsOptimizedInstances : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.stackId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'stackId'");
            }
            resourceInputs["allocatedMemory"] = args ? args.allocatedMemory : undefined;
            resourceInputs["autoAssignElasticIps"] = args ? args.autoAssignElasticIps : undefined;
            resourceInputs["autoAssignPublicIps"] = args ? args.autoAssignPublicIps : undefined;
            resourceInputs["autoHealing"] = args ? args.autoHealing : undefined;
            resourceInputs["cloudwatchConfiguration"] = args ? args.cloudwatchConfiguration : undefined;
            resourceInputs["customConfigureRecipes"] = args ? args.customConfigureRecipes : undefined;
            resourceInputs["customDeployRecipes"] = args ? args.customDeployRecipes : undefined;
            resourceInputs["customInstanceProfileArn"] = args ? args.customInstanceProfileArn : undefined;
            resourceInputs["customJson"] = args ? args.customJson : undefined;
            resourceInputs["customSecurityGroupIds"] = args ? args.customSecurityGroupIds : undefined;
            resourceInputs["customSetupRecipes"] = args ? args.customSetupRecipes : undefined;
            resourceInputs["customShutdownRecipes"] = args ? args.customShutdownRecipes : undefined;
            resourceInputs["customUndeployRecipes"] = args ? args.customUndeployRecipes : undefined;
            resourceInputs["drainElbOnShutdown"] = args ? args.drainElbOnShutdown : undefined;
            resourceInputs["ebsVolumes"] = args ? args.ebsVolumes : undefined;
            resourceInputs["elasticLoadBalancer"] = args ? args.elasticLoadBalancer : undefined;
            resourceInputs["installUpdatesOnBoot"] = args ? args.installUpdatesOnBoot : undefined;
            resourceInputs["instanceShutdownTimeout"] = args ? args.instanceShutdownTimeout : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["stackId"] = args ? args.stackId : undefined;
            resourceInputs["systemPackages"] = args ? args.systemPackages : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["useEbsOptimizedInstances"] = args ? args.useEbsOptimizedInstances : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MemcachedLayer.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing MemcachedLayer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MemcachedLayer(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MemcachedLayer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MemcachedLayer.__pulumiType;
    }
}
exports.MemcachedLayer = MemcachedLayer;
/** @internal */
MemcachedLayer.__pulumiType = 'aws:opsworks/memcachedLayer:MemcachedLayer';
