// @pulumi/aws/opsworks/userProfile.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an OpsWorks User Profile resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const myProfile = new aws.opsworks.UserProfile("myProfile", {
 *     userArn: aws_iam_user.user.arn,
 *     sshUsername: "my_user",
 * });
 * ```
 */
class UserProfile extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["allowSelfManagement"] = state ? state.allowSelfManagement : undefined;
            resourceInputs["sshPublicKey"] = state ? state.sshPublicKey : undefined;
            resourceInputs["sshUsername"] = state ? state.sshUsername : undefined;
            resourceInputs["userArn"] = state ? state.userArn : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.sshUsername === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sshUsername'");
            }
            if ((!args || args.userArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'userArn'");
            }
            resourceInputs["allowSelfManagement"] = args ? args.allowSelfManagement : undefined;
            resourceInputs["sshPublicKey"] = args ? args.sshPublicKey : undefined;
            resourceInputs["sshUsername"] = args ? args.sshUsername : undefined;
            resourceInputs["userArn"] = args ? args.userArn : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(UserProfile.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing UserProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new UserProfile(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of UserProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === UserProfile.__pulumiType;
    }
}
exports.UserProfile = UserProfile;
/** @internal */
UserProfile.__pulumiType = 'aws:opsworks/userProfile:UserProfile';
