// @pulumi/aws/organizations/organization.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Organization = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a resource to create an organization.
 *
 * !> **WARNING:** When migrating from a `featureSet` of `CONSOLIDATED_BILLING` to `ALL`, the Organization account owner will received an email stating the following: "You started the process to enable all features for your AWS organization. As part of that process, all member accounts that joined your organization by invitation must approve the change. You don’t need approval from member accounts that you directly created from within your AWS organization." After all member accounts have accepted the invitation, the Organization account owner must then finalize the changes via the [AWS Console](https://console.aws.amazon.com/organizations/home#/organization/settings/migration-progress). Until these steps are performed, the provider will perpetually show a difference, and the `DescribeOrganization` API will continue to show the `FeatureSet` as `CONSOLIDATED_BILLING`. See the [AWS Organizations documentation](https://docs.aws.amazon.com/organizations/latest/userguide/orgs_manage_org_support-all-features.html) for more information.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const org = new aws.organizations.Organization("org", {
 *     awsServiceAccessPrincipals: [
 *         "cloudtrail.amazonaws.com",
 *         "config.amazonaws.com",
 *     ],
 *     featureSet: "ALL",
 * });
 * ```
 *
 * ## Import
 *
 * The AWS organization can be imported by using the `id`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:organizations/organization:Organization my_org o-1234567
 * ```
 */
class Organization extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accounts"] = state ? state.accounts : undefined;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["awsServiceAccessPrincipals"] = state ? state.awsServiceAccessPrincipals : undefined;
            resourceInputs["enabledPolicyTypes"] = state ? state.enabledPolicyTypes : undefined;
            resourceInputs["featureSet"] = state ? state.featureSet : undefined;
            resourceInputs["masterAccountArn"] = state ? state.masterAccountArn : undefined;
            resourceInputs["masterAccountEmail"] = state ? state.masterAccountEmail : undefined;
            resourceInputs["masterAccountId"] = state ? state.masterAccountId : undefined;
            resourceInputs["nonMasterAccounts"] = state ? state.nonMasterAccounts : undefined;
            resourceInputs["roots"] = state ? state.roots : undefined;
        }
        else {
            const args = argsOrState;
            resourceInputs["awsServiceAccessPrincipals"] = args ? args.awsServiceAccessPrincipals : undefined;
            resourceInputs["enabledPolicyTypes"] = args ? args.enabledPolicyTypes : undefined;
            resourceInputs["featureSet"] = args ? args.featureSet : undefined;
            resourceInputs["accounts"] = undefined /*out*/;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["masterAccountArn"] = undefined /*out*/;
            resourceInputs["masterAccountEmail"] = undefined /*out*/;
            resourceInputs["masterAccountId"] = undefined /*out*/;
            resourceInputs["nonMasterAccounts"] = undefined /*out*/;
            resourceInputs["roots"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Organization.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Organization resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Organization(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Organization.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Organization.__pulumiType;
    }
}
exports.Organization = Organization;
/** @internal */
Organization.__pulumiType = 'aws:organizations/organization:Organization';
