// @pulumi/aws/detective/member.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Member = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a resource to manage an [Amazon Detective Member](https://docs.aws.amazon.com/detective/latest/APIReference/API_CreateMembers.html).
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const exampleGraph = new aws.detective.Graph("exampleGraph", {});
 * const exampleMember = new aws.detective.Member("exampleMember", {
 *     accountId: "AWS ACCOUNT ID",
 *     emailAddress: "EMAIL",
 *     graphArn: exampleGraph.id,
 *     message: "Message of the invitation",
 *     disableEmailNotification: true,
 * });
 * ```
 *
 * ## Import
 *
 * `aws_detective_member` can be imported using the ARN of the graph followed by the account ID of the member account, e.g.
 *
 * ```sh
 *  $ pulumi import aws:detective/member:Member example arn:aws:detective:us-east-1:123456789101:graph:231684d34gh74g4bae1dbc7bd807d02d/123456789012
 * ```
 */
class Member extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["administratorId"] = state ? state.administratorId : undefined;
            resourceInputs["disableEmailNotification"] = state ? state.disableEmailNotification : undefined;
            resourceInputs["disabledReason"] = state ? state.disabledReason : undefined;
            resourceInputs["emailAddress"] = state ? state.emailAddress : undefined;
            resourceInputs["graphArn"] = state ? state.graphArn : undefined;
            resourceInputs["invitedTime"] = state ? state.invitedTime : undefined;
            resourceInputs["message"] = state ? state.message : undefined;
            resourceInputs["status"] = state ? state.status : undefined;
            resourceInputs["updatedTime"] = state ? state.updatedTime : undefined;
            resourceInputs["volumeUsageInBytes"] = state ? state.volumeUsageInBytes : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.emailAddress === undefined) && !opts.urn) {
                throw new Error("Missing required property 'emailAddress'");
            }
            if ((!args || args.graphArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'graphArn'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["disableEmailNotification"] = args ? args.disableEmailNotification : undefined;
            resourceInputs["emailAddress"] = args ? args.emailAddress : undefined;
            resourceInputs["graphArn"] = args ? args.graphArn : undefined;
            resourceInputs["message"] = args ? args.message : undefined;
            resourceInputs["administratorId"] = undefined /*out*/;
            resourceInputs["disabledReason"] = undefined /*out*/;
            resourceInputs["invitedTime"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["updatedTime"] = undefined /*out*/;
            resourceInputs["volumeUsageInBytes"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Member.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Member resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Member(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Member.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Member.__pulumiType;
    }
}
exports.Member = Member;
/** @internal */
Member.__pulumiType = 'aws:detective/member:Member';
