// @pulumi/docker/registryImage.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegistryImage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * <!-- Bug: Type and Name are switched -->
 * Manages the lifecycle of docker image/tag in a registry means it can store one or more version of specific docker images and identified by their tags.
 *
 * ## Example Usage
 *
 * To be able to update an image itself when an updated image arrives.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as docker from "@pulumi/docker";
 * import * as process from "process";
 *
 * const helloworld = new docker.RegistryImage("helloworld", {
 *     build: {
 *         context: `${process.cwd()}/absolutePathToContextFolder`,
 *     },
 * });
 * ```
 */
class RegistryImage extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["build"] = state ? state.build : undefined;
            resourceInputs["insecureSkipVerify"] = state ? state.insecureSkipVerify : undefined;
            resourceInputs["keepRemotely"] = state ? state.keepRemotely : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["sha256Digest"] = state ? state.sha256Digest : undefined;
        }
        else {
            const args = argsOrState;
            resourceInputs["build"] = args ? args.build : undefined;
            resourceInputs["insecureSkipVerify"] = args ? args.insecureSkipVerify : undefined;
            resourceInputs["keepRemotely"] = args ? args.keepRemotely : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["sha256Digest"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RegistryImage.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing RegistryImage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RegistryImage(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RegistryImage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegistryImage.__pulumiType;
    }
}
exports.RegistryImage = RegistryImage;
/** @internal */
RegistryImage.__pulumiType = 'docker:index/registryImage:RegistryImage';
