// @pulumi/aws/emrcontainers/virtualCluster.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an EMR Containers (EMR on EKS) Virtual Cluster.
 *
 * ## Example Usage
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.emrcontainers.VirtualCluster("example", {containerProvider: {
 *     id: aws_eks_cluster.example.name,
 *     type: "EKS",
 *     info: {
 *         eksInfo: {
 *             namespace: "default",
 *         },
 *     },
 * }});
 * ```
 *
 * ## Import
 *
 * EKS Clusters can be imported using the `name`, e.g.
 *
 * ```sh
 *  $ pulumi import aws:emrcontainers/virtualCluster:VirtualCluster example
 * ```
 */
class VirtualCluster extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["containerProvider"] = state ? state.containerProvider : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.containerProvider === undefined) && !opts.urn) {
                throw new Error("Missing required property 'containerProvider'");
            }
            resourceInputs["containerProvider"] = args ? args.containerProvider : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["tagsAll"] = args ? args.tagsAll : undefined;
            resourceInputs["arn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VirtualCluster.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing VirtualCluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VirtualCluster(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualCluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualCluster.__pulumiType;
    }
}
exports.VirtualCluster = VirtualCluster;
/** @internal */
VirtualCluster.__pulumiType = 'aws:emrcontainers/virtualCluster:VirtualCluster';
