// @pulumi/aws/route53domains/registeredDomain.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegisteredDomain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.route53domains.RegisteredDomain("example", {
 *     domainName: "example.com",
 *     nameServers: [
 *         {
 *             name: "ns-195.awsdns-24.com",
 *         },
 *         {
 *             name: "ns-874.awsdns-45.net",
 *         },
 *     ],
 *     tags: {
 *         Environment: "test",
 *     },
 * });
 * ```
 */
class RegisteredDomain extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["abuseContactEmail"] = state ? state.abuseContactEmail : undefined;
            resourceInputs["abuseContactPhone"] = state ? state.abuseContactPhone : undefined;
            resourceInputs["adminContact"] = state ? state.adminContact : undefined;
            resourceInputs["adminPrivacy"] = state ? state.adminPrivacy : undefined;
            resourceInputs["autoRenew"] = state ? state.autoRenew : undefined;
            resourceInputs["creationDate"] = state ? state.creationDate : undefined;
            resourceInputs["domainName"] = state ? state.domainName : undefined;
            resourceInputs["expirationDate"] = state ? state.expirationDate : undefined;
            resourceInputs["nameServers"] = state ? state.nameServers : undefined;
            resourceInputs["registrantContact"] = state ? state.registrantContact : undefined;
            resourceInputs["registrantPrivacy"] = state ? state.registrantPrivacy : undefined;
            resourceInputs["registrarName"] = state ? state.registrarName : undefined;
            resourceInputs["registrarUrl"] = state ? state.registrarUrl : undefined;
            resourceInputs["reseller"] = state ? state.reseller : undefined;
            resourceInputs["statusLists"] = state ? state.statusLists : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
            resourceInputs["techContact"] = state ? state.techContact : undefined;
            resourceInputs["techPrivacy"] = state ? state.techPrivacy : undefined;
            resourceInputs["transferLock"] = state ? state.transferLock : undefined;
            resourceInputs["updatedDate"] = state ? state.updatedDate : undefined;
            resourceInputs["whoisServer"] = state ? state.whoisServer : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.domainName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'domainName'");
            }
            resourceInputs["adminContact"] = args ? args.adminContact : undefined;
            resourceInputs["adminPrivacy"] = args ? args.adminPrivacy : undefined;
            resourceInputs["autoRenew"] = args ? args.autoRenew : undefined;
            resourceInputs["domainName"] = args ? args.domainName : undefined;
            resourceInputs["nameServers"] = args ? args.nameServers : undefined;
            resourceInputs["registrantContact"] = args ? args.registrantContact : undefined;
            resourceInputs["registrantPrivacy"] = args ? args.registrantPrivacy : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["tagsAll"] = args ? args.tagsAll : undefined;
            resourceInputs["techContact"] = args ? args.techContact : undefined;
            resourceInputs["techPrivacy"] = args ? args.techPrivacy : undefined;
            resourceInputs["transferLock"] = args ? args.transferLock : undefined;
            resourceInputs["abuseContactEmail"] = undefined /*out*/;
            resourceInputs["abuseContactPhone"] = undefined /*out*/;
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["expirationDate"] = undefined /*out*/;
            resourceInputs["registrarName"] = undefined /*out*/;
            resourceInputs["registrarUrl"] = undefined /*out*/;
            resourceInputs["reseller"] = undefined /*out*/;
            resourceInputs["statusLists"] = undefined /*out*/;
            resourceInputs["updatedDate"] = undefined /*out*/;
            resourceInputs["whoisServer"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RegisteredDomain.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing RegisteredDomain resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RegisteredDomain(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RegisteredDomain.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegisteredDomain.__pulumiType;
    }
}
exports.RegisteredDomain = RegisteredDomain;
/** @internal */
RegisteredDomain.__pulumiType = 'aws:route53domains/registeredDomain:RegisteredDomain';
