// @pulumi/aws/iam/virtualMfaDevice.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualMfaDevice = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 *
 * **Using certs on file:**
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.iam.VirtualMfaDevice("example", {
 *     virtualMfaDeviceName: "example",
 * });
 * ```
 *
 * ## Import
 *
 * IAM Virtual MFA Devices can be imported using the `arn`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:iam/virtualMfaDevice:VirtualMfaDevice example arn:aws:iam::123456789012:mfa/example
 * ```
 */
class VirtualMfaDevice extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["base32StringSeed"] = state ? state.base32StringSeed : undefined;
            resourceInputs["path"] = state ? state.path : undefined;
            resourceInputs["qrCodePng"] = state ? state.qrCodePng : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
            resourceInputs["virtualMfaDeviceName"] = state ? state.virtualMfaDeviceName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.virtualMfaDeviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'virtualMfaDeviceName'");
            }
            resourceInputs["path"] = args ? args.path : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["virtualMfaDeviceName"] = args ? args.virtualMfaDeviceName : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["base32StringSeed"] = undefined /*out*/;
            resourceInputs["qrCodePng"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VirtualMfaDevice.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing VirtualMfaDevice resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VirtualMfaDevice(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualMfaDevice.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualMfaDevice.__pulumiType;
    }
}
exports.VirtualMfaDevice = VirtualMfaDevice;
/** @internal */
VirtualMfaDevice.__pulumiType = 'aws:iam/virtualMfaDevice:VirtualMfaDevice';
