// @pulumi/aws/fms/policy.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Policy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a resource to create an AWS Firewall Manager policy. You need to be using AWS organizations and have enabled the Firewall Manager administrator account.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const exampleRuleGroup = new aws.wafregional.RuleGroup("exampleRuleGroup", {metricName: "WAFRuleGroupExample"});
 * const examplePolicy = new aws.fms.Policy("examplePolicy", {
 *     excludeResourceTags: false,
 *     remediationEnabled: false,
 *     resourceTypeLists: ["AWS::ElasticLoadBalancingV2::LoadBalancer"],
 *     securityServicePolicyData: {
 *         type: "WAF",
 *         managedServiceData: exampleRuleGroup.id.apply(id => JSON.stringify({
 *             type: "WAF",
 *             ruleGroups: [{
 *                 id: id,
 *                 overrideAction: {
 *                     type: "COUNT",
 *                 },
 *             }],
 *             defaultAction: {
 *                 type: "BLOCK",
 *             },
 *             overrideCustomerWebACLAssociation: false,
 *         })),
 *     },
 *     tags: {
 *         Name: "example-fms-policy",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Firewall Manager policies can be imported using the policy ID, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:fms/policy:Policy example 5be49585-a7e3-4c49-dde1-a179fe4a619a
 * ```
 */
class Policy extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["deleteAllPolicyResources"] = state ? state.deleteAllPolicyResources : undefined;
            resourceInputs["deleteUnusedFmManagedResources"] = state ? state.deleteUnusedFmManagedResources : undefined;
            resourceInputs["excludeMap"] = state ? state.excludeMap : undefined;
            resourceInputs["excludeResourceTags"] = state ? state.excludeResourceTags : undefined;
            resourceInputs["includeMap"] = state ? state.includeMap : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["policyUpdateToken"] = state ? state.policyUpdateToken : undefined;
            resourceInputs["remediationEnabled"] = state ? state.remediationEnabled : undefined;
            resourceInputs["resourceTags"] = state ? state.resourceTags : undefined;
            resourceInputs["resourceType"] = state ? state.resourceType : undefined;
            resourceInputs["resourceTypeLists"] = state ? state.resourceTypeLists : undefined;
            resourceInputs["securityServicePolicyData"] = state ? state.securityServicePolicyData : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.excludeResourceTags === undefined) && !opts.urn) {
                throw new Error("Missing required property 'excludeResourceTags'");
            }
            if ((!args || args.securityServicePolicyData === undefined) && !opts.urn) {
                throw new Error("Missing required property 'securityServicePolicyData'");
            }
            resourceInputs["deleteAllPolicyResources"] = args ? args.deleteAllPolicyResources : undefined;
            resourceInputs["deleteUnusedFmManagedResources"] = args ? args.deleteUnusedFmManagedResources : undefined;
            resourceInputs["excludeMap"] = args ? args.excludeMap : undefined;
            resourceInputs["excludeResourceTags"] = args ? args.excludeResourceTags : undefined;
            resourceInputs["includeMap"] = args ? args.includeMap : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["remediationEnabled"] = args ? args.remediationEnabled : undefined;
            resourceInputs["resourceTags"] = args ? args.resourceTags : undefined;
            resourceInputs["resourceType"] = args ? args.resourceType : undefined;
            resourceInputs["resourceTypeLists"] = args ? args.resourceTypeLists : undefined;
            resourceInputs["securityServicePolicyData"] = args ? args.securityServicePolicyData : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["tagsAll"] = args ? args.tagsAll : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["policyUpdateToken"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Policy.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Policy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Policy(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Policy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Policy.__pulumiType;
    }
}
exports.Policy = Policy;
/** @internal */
Policy.__pulumiType = 'aws:fms/policy:Policy';
