// @pulumi/aws/devicefarm/networkProfile.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a resource to manage AWS Device Farm Network Profiles.
 * ∂
 * > **NOTE:** AWS currently has limited regional support for Device Farm (e.g., `us-west-2`). See [AWS Device Farm endpoints and quotas](https://docs.aws.amazon.com/general/latest/gr/devicefarm.html) for information on supported regions.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const exampleProject = new aws.devicefarm.Project("exampleProject", {});
 * const exampleNetworkProfile = new aws.devicefarm.NetworkProfile("exampleNetworkProfile", {projectArn: exampleProject.arn});
 * ```
 *
 * ## Import
 *
 * DeviceFarm Network Profiles can be imported by their arn
 *
 * ```sh
 *  $ pulumi import aws:devicefarm/networkProfile:NetworkProfile example arn:aws:devicefarm:us-west-2:123456789012:networkprofile:4fa784c7-ccb4-4dbf-ba4f-02198320daa1
 * ```
 */
class NetworkProfile extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["downlinkBandwidthBits"] = state ? state.downlinkBandwidthBits : undefined;
            resourceInputs["downlinkDelayMs"] = state ? state.downlinkDelayMs : undefined;
            resourceInputs["downlinkJitterMs"] = state ? state.downlinkJitterMs : undefined;
            resourceInputs["downlinkLossPercent"] = state ? state.downlinkLossPercent : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["projectArn"] = state ? state.projectArn : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
            resourceInputs["type"] = state ? state.type : undefined;
            resourceInputs["uplinkBandwidthBits"] = state ? state.uplinkBandwidthBits : undefined;
            resourceInputs["uplinkDelayMs"] = state ? state.uplinkDelayMs : undefined;
            resourceInputs["uplinkJitterMs"] = state ? state.uplinkJitterMs : undefined;
            resourceInputs["uplinkLossPercent"] = state ? state.uplinkLossPercent : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.projectArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'projectArn'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["downlinkBandwidthBits"] = args ? args.downlinkBandwidthBits : undefined;
            resourceInputs["downlinkDelayMs"] = args ? args.downlinkDelayMs : undefined;
            resourceInputs["downlinkJitterMs"] = args ? args.downlinkJitterMs : undefined;
            resourceInputs["downlinkLossPercent"] = args ? args.downlinkLossPercent : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["projectArn"] = args ? args.projectArn : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["uplinkBandwidthBits"] = args ? args.uplinkBandwidthBits : undefined;
            resourceInputs["uplinkDelayMs"] = args ? args.uplinkDelayMs : undefined;
            resourceInputs["uplinkJitterMs"] = args ? args.uplinkJitterMs : undefined;
            resourceInputs["uplinkLossPercent"] = args ? args.uplinkLossPercent : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(NetworkProfile.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing NetworkProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new NetworkProfile(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkProfile.__pulumiType;
    }
}
exports.NetworkProfile = NetworkProfile;
/** @internal */
NetworkProfile.__pulumiType = 'aws:devicefarm/networkProfile:NetworkProfile';
