// @pulumi/aws/apprunner/vpcConnector.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an App Runner VPC Connector.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const connector = new aws.apprunner.VpcConnector("connector", {
 *     securityGroups: [
 *         "sg1",
 *         "sg2",
 *     ],
 *     subnets: [
 *         "subnet1",
 *         "subnet2",
 *     ],
 *     vpcConnectorName: "name",
 * });
 * ```
 *
 * ## Import
 *
 * App Runner vpc connector can be imported by using the `vpc_connector_name`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:apprunner/vpcConnector:VpcConnector vpc_connector_name
 * ```
 */
class VpcConnector extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["securityGroups"] = state ? state.securityGroups : undefined;
            resourceInputs["status"] = state ? state.status : undefined;
            resourceInputs["subnets"] = state ? state.subnets : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["vpcConnectorName"] = state ? state.vpcConnectorName : undefined;
            resourceInputs["vpcConnectorRevision"] = state ? state.vpcConnectorRevision : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.securityGroups === undefined) && !opts.urn) {
                throw new Error("Missing required property 'securityGroups'");
            }
            if ((!args || args.subnets === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subnets'");
            }
            if ((!args || args.vpcConnectorName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vpcConnectorName'");
            }
            resourceInputs["securityGroups"] = args ? args.securityGroups : undefined;
            resourceInputs["subnets"] = args ? args.subnets : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["vpcConnectorName"] = args ? args.vpcConnectorName : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["vpcConnectorRevision"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VpcConnector.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing VpcConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VpcConnector(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VpcConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VpcConnector.__pulumiType;
    }
}
exports.VpcConnector = VpcConnector;
/** @internal */
VpcConnector.__pulumiType = 'aws:apprunner/vpcConnector:VpcConnector';
