// @pulumi/aws/applicationloadbalancing/targetGroupAttachment.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetGroupAttachment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides the ability to register instances and containers with an Application Load Balancer (ALB) or Network Load Balancer (NLB) target group. For attaching resources with Elastic Load Balancer (ELB), see the `aws.elb.Attachment` resource.
 *
 * > **Note:** `aws.alb.TargetGroupAttachment` is known as `aws.lb.TargetGroupAttachment`. The functionality is identical.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const testTargetGroup = new aws.lb.TargetGroup("testTargetGroup", {});
 * // ... other configuration ...
 * const testInstance = new aws.ec2.Instance("testInstance", {});
 * // ... other configuration ...
 * const testTargetGroupAttachment = new aws.lb.TargetGroupAttachment("testTargetGroupAttachment", {
 *     targetGroupArn: testTargetGroup.arn,
 *     targetId: testInstance.id,
 *     port: 80,
 * });
 * ```
 * ## Usage with lambda
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const testTargetGroup = new aws.lb.TargetGroup("testTargetGroup", {targetType: "lambda"});
 * const testFunction = new aws.lambda.Function("testFunction", {});
 * // ... other configuration ...
 * const withLb = new aws.lambda.Permission("withLb", {
 *     action: "lambda:InvokeFunction",
 *     "function": testFunction.arn,
 *     principal: "elasticloadbalancing.amazonaws.com",
 *     sourceArn: testTargetGroup.arn,
 * });
 * const testTargetGroupAttachment = new aws.lb.TargetGroupAttachment("testTargetGroupAttachment", {
 *     targetGroupArn: testTargetGroup.arn,
 *     targetId: testFunction.arn,
 * }, {
 *     dependsOn: [withLb],
 * });
 * ```
 *
 * ## Import
 *
 * Target Group Attachments cannot be imported.
 *
 * @deprecated aws.applicationloadbalancing.TargetGroupAttachment has been deprecated in favor of aws.alb.TargetGroupAttachment
 */
class TargetGroupAttachment extends pulumi.CustomResource {
    /** @deprecated aws.applicationloadbalancing.TargetGroupAttachment has been deprecated in favor of aws.alb.TargetGroupAttachment */
    constructor(name, argsOrState, opts) {
        pulumi.log.warn("TargetGroupAttachment is deprecated: aws.applicationloadbalancing.TargetGroupAttachment has been deprecated in favor of aws.alb.TargetGroupAttachment");
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["availabilityZone"] = state ? state.availabilityZone : undefined;
            resourceInputs["port"] = state ? state.port : undefined;
            resourceInputs["targetGroupArn"] = state ? state.targetGroupArn : undefined;
            resourceInputs["targetId"] = state ? state.targetId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.targetGroupArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetGroupArn'");
            }
            if ((!args || args.targetId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetId'");
            }
            resourceInputs["availabilityZone"] = args ? args.availabilityZone : undefined;
            resourceInputs["port"] = args ? args.port : undefined;
            resourceInputs["targetGroupArn"] = args ? args.targetGroupArn : undefined;
            resourceInputs["targetId"] = args ? args.targetId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TargetGroupAttachment.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing TargetGroupAttachment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        pulumi.log.warn("TargetGroupAttachment is deprecated: aws.applicationloadbalancing.TargetGroupAttachment has been deprecated in favor of aws.alb.TargetGroupAttachment");
        return new TargetGroupAttachment(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TargetGroupAttachment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TargetGroupAttachment.__pulumiType;
    }
}
exports.TargetGroupAttachment = TargetGroupAttachment;
/** @internal */
TargetGroupAttachment.__pulumiType = 'aws:applicationloadbalancing/targetGroupAttachment:TargetGroupAttachment';
