// @pulumi/aws/route53recoverycontrol/safetyRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SafetyRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an AWS Route 53 Recovery Control Config Safety Rule
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.route53recoverycontrol.SafetyRule("example", {
 *     assertedControls: [aws_route53recoverycontrolconfig_routing_control.example.arn],
 *     controlPanelArn: "arn:aws:route53-recovery-control::313517334327:controlpanel/abd5fbfc052d4844a082dbf400f61da8",
 *     waitPeriodMs: 5000,
 *     ruleConfig: {
 *         inverted: false,
 *         threshold: 1,
 *         type: "ATLEAST",
 *     },
 * });
 * ```
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.route53recoverycontrol.SafetyRule("example", {
 *     controlPanelArn: "arn:aws:route53-recovery-control::313517334327:controlpanel/abd5fbfc052d4844a082dbf400f61da8",
 *     waitPeriodMs: 5000,
 *     gatingControls: [aws_route53recoverycontrolconfig_routing_control.example.arn],
 *     targetControls: [aws_route53recoverycontrolconfig_routing_control.example.arn],
 *     ruleConfig: {
 *         inverted: false,
 *         threshold: 1,
 *         type: "ATLEAST",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Route53 Recovery Control Config Safety Rule can be imported via the safety rule ARN, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:route53recoverycontrol/safetyRule:SafetyRule myrule arn:aws:route53-recovery-control::313517334327:controlpanel/1bfba17df8684f5dab0467b71424f7e8/safetyrule/3bacc77003364c0f
 * ```
 */
class SafetyRule extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["assertedControls"] = state ? state.assertedControls : undefined;
            resourceInputs["controlPanelArn"] = state ? state.controlPanelArn : undefined;
            resourceInputs["gatingControls"] = state ? state.gatingControls : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["ruleConfig"] = state ? state.ruleConfig : undefined;
            resourceInputs["status"] = state ? state.status : undefined;
            resourceInputs["targetControls"] = state ? state.targetControls : undefined;
            resourceInputs["waitPeriodMs"] = state ? state.waitPeriodMs : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.controlPanelArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'controlPanelArn'");
            }
            if ((!args || args.ruleConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ruleConfig'");
            }
            if ((!args || args.waitPeriodMs === undefined) && !opts.urn) {
                throw new Error("Missing required property 'waitPeriodMs'");
            }
            resourceInputs["assertedControls"] = args ? args.assertedControls : undefined;
            resourceInputs["controlPanelArn"] = args ? args.controlPanelArn : undefined;
            resourceInputs["gatingControls"] = args ? args.gatingControls : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["ruleConfig"] = args ? args.ruleConfig : undefined;
            resourceInputs["targetControls"] = args ? args.targetControls : undefined;
            resourceInputs["waitPeriodMs"] = args ? args.waitPeriodMs : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SafetyRule.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SafetyRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SafetyRule(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SafetyRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SafetyRule.__pulumiType;
    }
}
exports.SafetyRule = SafetyRule;
/** @internal */
SafetyRule.__pulumiType = 'aws:route53recoverycontrol/safetyRule:SafetyRule';
