// @pulumi/aws/efs/replicationConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Creates a replica of an existing EFS file system in the same or another region. Creating this resource causes the source EFS file system to be replicated to a new read-only destination EFS file system. Deleting this resource will cause the replication from source to destination to stop and the destination file system will no longer be read only.
 *
 * > **NOTE:** Deleting this resource does **not** delete the destination file system that was created.
 *
 * ## Example Usage
 *
 * Will create a replica using regional storage in us-west-2 that will be encrypted by the default EFS KMS key `/aws/elasticfilesystem`.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const exampleFileSystem = new aws.efs.FileSystem("exampleFileSystem", {});
 * const exampleReplicationConfiguration = new aws.efs.ReplicationConfiguration("exampleReplicationConfiguration", {
 *     sourceFileSystemId: exampleFileSystem.id,
 *     destination: {
 *         region: "us-west-2",
 *     },
 * });
 * ```
 *
 * Replica will be created as One Zone storage in the us-west-2b Availability Zone and encrypted with the specified KMS key.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const exampleFileSystem = new aws.efs.FileSystem("exampleFileSystem", {});
 * const exampleReplicationConfiguration = new aws.efs.ReplicationConfiguration("exampleReplicationConfiguration", {
 *     sourceFileSystemId: exampleFileSystem.id,
 *     destination: {
 *         availabilityZoneName: "us-west-2b",
 *         kmsKeyId: "1234abcd-12ab-34cd-56ef-1234567890ab",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * EFS Replication Configurations can be imported using the file system ID of either the source or destination file system. When importing, the `availability_zone_name` and `kms_key_id` attributes must **not** be set in the configuration. The AWS API does not return these values when querying the replication configuration and their presence will therefore show as a diff in a subsequent plan.
 *
 * ```sh
 *  $ pulumi import aws:efs/replicationConfiguration:ReplicationConfiguration example fs-id
 * ```
 */
class ReplicationConfiguration extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["creationTime"] = state ? state.creationTime : undefined;
            resourceInputs["destination"] = state ? state.destination : undefined;
            resourceInputs["originalSourceFileSystemArn"] = state ? state.originalSourceFileSystemArn : undefined;
            resourceInputs["sourceFileSystemArn"] = state ? state.sourceFileSystemArn : undefined;
            resourceInputs["sourceFileSystemId"] = state ? state.sourceFileSystemId : undefined;
            resourceInputs["sourceFileSystemRegion"] = state ? state.sourceFileSystemRegion : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.destination === undefined) && !opts.urn) {
                throw new Error("Missing required property 'destination'");
            }
            if ((!args || args.sourceFileSystemId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceFileSystemId'");
            }
            resourceInputs["destination"] = args ? args.destination : undefined;
            resourceInputs["sourceFileSystemId"] = args ? args.sourceFileSystemId : undefined;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["originalSourceFileSystemArn"] = undefined /*out*/;
            resourceInputs["sourceFileSystemArn"] = undefined /*out*/;
            resourceInputs["sourceFileSystemRegion"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ReplicationConfiguration.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ReplicationConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ReplicationConfiguration(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ReplicationConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReplicationConfiguration.__pulumiType;
    }
}
exports.ReplicationConfiguration = ReplicationConfiguration;
/** @internal */
ReplicationConfiguration.__pulumiType = 'aws:efs/replicationConfiguration:ReplicationConfiguration';
