// @pulumi/aws/worklink/fleet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Fleet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 *
 * Basic usage:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.worklink.Fleet("example", {});
 * ```
 *
 * Network Configuration Usage:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.worklink.Fleet("example", {network: {
 *     vpcId: aws_vpc.test.id,
 *     subnetIds: [aws_subnet.test.map(__item => __item.id)],
 *     securityGroupIds: [aws_security_group.test.id],
 * }});
 * ```
 *
 * Identity Provider Configuration Usage:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 * import * from "fs";
 *
 * const test = new aws.worklink.Fleet("test", {identityProvider: {
 *     type: "SAML",
 *     samlMetadata: fs.readFileSync("saml-metadata.xml"),
 * }});
 * ```
 *
 * ## Import
 *
 * WorkLink can be imported using the ARN, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:worklink/fleet:Fleet test arn:aws:worklink::123456789012:fleet/example
 * ```
 */
class Fleet extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["auditStreamArn"] = state ? state.auditStreamArn : undefined;
            resourceInputs["companyCode"] = state ? state.companyCode : undefined;
            resourceInputs["createdTime"] = state ? state.createdTime : undefined;
            resourceInputs["deviceCaCertificate"] = state ? state.deviceCaCertificate : undefined;
            resourceInputs["displayName"] = state ? state.displayName : undefined;
            resourceInputs["identityProvider"] = state ? state.identityProvider : undefined;
            resourceInputs["lastUpdatedTime"] = state ? state.lastUpdatedTime : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["network"] = state ? state.network : undefined;
            resourceInputs["optimizeForEndUserLocation"] = state ? state.optimizeForEndUserLocation : undefined;
        }
        else {
            const args = argsOrState;
            resourceInputs["auditStreamArn"] = args ? args.auditStreamArn : undefined;
            resourceInputs["deviceCaCertificate"] = args ? args.deviceCaCertificate : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["identityProvider"] = args ? args.identityProvider : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["network"] = args ? args.network : undefined;
            resourceInputs["optimizeForEndUserLocation"] = args ? args.optimizeForEndUserLocation : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["companyCode"] = undefined /*out*/;
            resourceInputs["createdTime"] = undefined /*out*/;
            resourceInputs["lastUpdatedTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Fleet.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Fleet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Fleet(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Fleet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Fleet.__pulumiType;
    }
}
exports.Fleet = Fleet;
/** @internal */
Fleet.__pulumiType = 'aws:worklink/fleet:Fleet';
