// @pulumi/aws/datasync/locationHdfs.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocationHdfs = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an HDFS Location within AWS DataSync.
 *
 * > **NOTE:** The DataSync Agents must be available before creating this resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.datasync.LocationHdfs("example", {
 *     agentArns: [aws_datasync_agent.example.arn],
 *     authenticationType: "SIMPLE",
 *     simpleUser: "example",
 *     nameNodes: [{
 *         hostname: aws_instance.example.private_dns,
 *         port: 80,
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * `aws_datasync_location_hdfs` can be imported by using the Amazon Resource Name (ARN), e.g.,
 *
 * ```sh
 *  $ pulumi import aws:datasync/locationHdfs:LocationHdfs example arn:aws:datasync:us-east-1:123456789012:location/loc-12345678901234567
 * ```
 */
class LocationHdfs extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["agentArns"] = state ? state.agentArns : undefined;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["authenticationType"] = state ? state.authenticationType : undefined;
            resourceInputs["blockSize"] = state ? state.blockSize : undefined;
            resourceInputs["kerberosKeytab"] = state ? state.kerberosKeytab : undefined;
            resourceInputs["kerberosKrb5Conf"] = state ? state.kerberosKrb5Conf : undefined;
            resourceInputs["kerberosPrincipal"] = state ? state.kerberosPrincipal : undefined;
            resourceInputs["kmsKeyProviderUri"] = state ? state.kmsKeyProviderUri : undefined;
            resourceInputs["nameNodes"] = state ? state.nameNodes : undefined;
            resourceInputs["qopConfiguration"] = state ? state.qopConfiguration : undefined;
            resourceInputs["replicationFactor"] = state ? state.replicationFactor : undefined;
            resourceInputs["simpleUser"] = state ? state.simpleUser : undefined;
            resourceInputs["subdirectory"] = state ? state.subdirectory : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
            resourceInputs["uri"] = state ? state.uri : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.agentArns === undefined) && !opts.urn) {
                throw new Error("Missing required property 'agentArns'");
            }
            if ((!args || args.nameNodes === undefined) && !opts.urn) {
                throw new Error("Missing required property 'nameNodes'");
            }
            resourceInputs["agentArns"] = args ? args.agentArns : undefined;
            resourceInputs["authenticationType"] = args ? args.authenticationType : undefined;
            resourceInputs["blockSize"] = args ? args.blockSize : undefined;
            resourceInputs["kerberosKeytab"] = args ? args.kerberosKeytab : undefined;
            resourceInputs["kerberosKrb5Conf"] = args ? args.kerberosKrb5Conf : undefined;
            resourceInputs["kerberosPrincipal"] = args ? args.kerberosPrincipal : undefined;
            resourceInputs["kmsKeyProviderUri"] = args ? args.kmsKeyProviderUri : undefined;
            resourceInputs["nameNodes"] = args ? args.nameNodes : undefined;
            resourceInputs["qopConfiguration"] = args ? args.qopConfiguration : undefined;
            resourceInputs["replicationFactor"] = args ? args.replicationFactor : undefined;
            resourceInputs["simpleUser"] = args ? args.simpleUser : undefined;
            resourceInputs["subdirectory"] = args ? args.subdirectory : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
            resourceInputs["uri"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LocationHdfs.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing LocationHdfs resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LocationHdfs(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LocationHdfs.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LocationHdfs.__pulumiType;
    }
}
exports.LocationHdfs = LocationHdfs;
/** @internal */
LocationHdfs.__pulumiType = 'aws:datasync/locationHdfs:LocationHdfs';
