// @pulumi/aws/kinesis/streamConsumer.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamConsumer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a resource to manage a Kinesis Stream Consumer.
 *
 * > **Note:** You can register up to 20 consumers per stream. A given consumer can only be registered with one stream at a time.
 *
 * For more details, see the [Amazon Kinesis Stream Consumer Documentation](https://docs.aws.amazon.com/streams/latest/dev/amazon-kinesis-consumers.html).
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const exampleStream = new aws.kinesis.Stream("exampleStream", {shardCount: 1});
 * const exampleStreamConsumer = new aws.kinesis.StreamConsumer("exampleStreamConsumer", {streamArn: exampleStream.arn});
 * ```
 *
 * ## Import
 *
 * Kinesis Stream Consumers can be imported using the Amazon Resource Name (ARN) e.g.,
 *
 * ```sh
 *  $ pulumi import aws:kinesis/streamConsumer:StreamConsumer example arn:aws:kinesis:us-west-2:123456789012:stream/example/consumer/example:1616044553
 * ```
 *
 *  [1]https://docs.aws.amazon.com/streams/latest/dev/amazon-kinesis-consumers.html
 */
class StreamConsumer extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["creationTimestamp"] = state ? state.creationTimestamp : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["streamArn"] = state ? state.streamArn : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.streamArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'streamArn'");
            }
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["streamArn"] = args ? args.streamArn : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["creationTimestamp"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(StreamConsumer.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing StreamConsumer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new StreamConsumer(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StreamConsumer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StreamConsumer.__pulumiType;
    }
}
exports.StreamConsumer = StreamConsumer;
/** @internal */
StreamConsumer.__pulumiType = 'aws:kinesis/streamConsumer:StreamConsumer';
