// @pulumi/aws/networkmanager/transitGatewayConnectPeerAssociation.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransitGatewayConnectPeerAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Associates a transit gateway Connect peer with a device, and optionally, with a link.
 * If you specify a link, it must be associated with the specified device.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.networkmanager.TransitGatewayConnectPeerAssociation("example", {
 *     globalNetworkId: aws_networkmanager_global_network.example.id,
 *     deviceId: aws_networkmanager_device.example.id,
 *     transitGatewayConnectPeerArn: aws_ec2_transit_gateway_connect_peer.example.arn,
 * });
 * ```
 *
 * ## Import
 *
 * `aws_networkmanager_transit_gateway_connect_peer_association` can be imported using the global network ID and customer gateway ARN, e.g.
 *
 * ```sh
 *  $ pulumi import aws:networkmanager/transitGatewayConnectPeerAssociation:TransitGatewayConnectPeerAssociation example global-network-0d47f6t230mz46dy4,arn:aws:ec2:us-west-2:123456789012:transit-gateway-connect-peer/tgw-connect-peer-12345678
 * ```
 */
class TransitGatewayConnectPeerAssociation extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["deviceId"] = state ? state.deviceId : undefined;
            resourceInputs["globalNetworkId"] = state ? state.globalNetworkId : undefined;
            resourceInputs["linkId"] = state ? state.linkId : undefined;
            resourceInputs["transitGatewayConnectPeerArn"] = state ? state.transitGatewayConnectPeerArn : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.deviceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deviceId'");
            }
            if ((!args || args.globalNetworkId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'globalNetworkId'");
            }
            if ((!args || args.transitGatewayConnectPeerArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'transitGatewayConnectPeerArn'");
            }
            resourceInputs["deviceId"] = args ? args.deviceId : undefined;
            resourceInputs["globalNetworkId"] = args ? args.globalNetworkId : undefined;
            resourceInputs["linkId"] = args ? args.linkId : undefined;
            resourceInputs["transitGatewayConnectPeerArn"] = args ? args.transitGatewayConnectPeerArn : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TransitGatewayConnectPeerAssociation.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing TransitGatewayConnectPeerAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TransitGatewayConnectPeerAssociation(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TransitGatewayConnectPeerAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TransitGatewayConnectPeerAssociation.__pulumiType;
    }
}
exports.TransitGatewayConnectPeerAssociation = TransitGatewayConnectPeerAssociation;
/** @internal */
TransitGatewayConnectPeerAssociation.__pulumiType = 'aws:networkmanager/transitGatewayConnectPeerAssociation:TransitGatewayConnectPeerAssociation';
