// @pulumi/aws/ses/configurationSet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigurationSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an SES configuration set resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.ses.ConfigurationSet("test", {});
 * ```
 * ### Require TLS Connections
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.ses.ConfigurationSet("test", {
 *     deliveryOptions: {
 *         tlsPolicy: "Require",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * SES Configuration Sets can be imported using their `name`, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:ses/configurationSet:ConfigurationSet test some-configuration-set-test
 * ```
 */
class ConfigurationSet extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["deliveryOptions"] = state ? state.deliveryOptions : undefined;
            resourceInputs["lastFreshStart"] = state ? state.lastFreshStart : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["reputationMetricsEnabled"] = state ? state.reputationMetricsEnabled : undefined;
            resourceInputs["sendingEnabled"] = state ? state.sendingEnabled : undefined;
        }
        else {
            const args = argsOrState;
            resourceInputs["deliveryOptions"] = args ? args.deliveryOptions : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["reputationMetricsEnabled"] = args ? args.reputationMetricsEnabled : undefined;
            resourceInputs["sendingEnabled"] = args ? args.sendingEnabled : undefined;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["lastFreshStart"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "aws:ses/confgurationSet:ConfgurationSet" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ConfigurationSet.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ConfigurationSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ConfigurationSet(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ConfigurationSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ConfigurationSet.__pulumiType;
    }
}
exports.ConfigurationSet = ConfigurationSet;
/** @internal */
ConfigurationSet.__pulumiType = 'aws:ses/configurationSet:ConfigurationSet';
