// @pulumi/aws/ses/template.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Template = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a resource to create a SES template.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const myTemplate = new aws.ses.Template("MyTemplate", {
 *     html: "<h1>Hello {{name}},</h1><p>Your favorite animal is {{favoriteanimal}}.</p>",
 *     subject: "Greetings, {{name}}!",
 *     text: `Hello {{name}},
 * Your favorite animal is {{favoriteanimal}}.`,
 * });
 * ```
 *
 * ## Import
 *
 * SES templates can be imported using the template name, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:ses/template:Template MyTemplate MyTemplate
 * ```
 */
class Template extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["html"] = state ? state.html : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["subject"] = state ? state.subject : undefined;
            resourceInputs["text"] = state ? state.text : undefined;
        }
        else {
            const args = argsOrState;
            resourceInputs["html"] = args ? args.html : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["subject"] = args ? args.subject : undefined;
            resourceInputs["text"] = args ? args.text : undefined;
            resourceInputs["arn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Template.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Template resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Template(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Template.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Template.__pulumiType;
    }
}
exports.Template = Template;
/** @internal */
Template.__pulumiType = 'aws:ses/template:Template';
