// @pulumi/aws/codeartifact/repository.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Repository = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a CodeArtifact Repository Resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const exampleKey = new aws.kms.Key("exampleKey", {description: "domain key"});
 * const exampleDomain = new aws.codeartifact.Domain("exampleDomain", {
 *     domain: "example",
 *     encryptionKey: exampleKey.arn,
 * });
 * const test = new aws.codeartifact.Repository("test", {
 *     repository: "example",
 *     domain: exampleDomain.domain,
 * });
 * ```
 * ### With Upstream Repository
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const upstream = new aws.codeartifact.Repository("upstream", {
 *     repository: "upstream",
 *     domain: aws_codeartifact_domain.test.domain,
 * });
 * const test = new aws.codeartifact.Repository("test", {
 *     repository: "example",
 *     domain: aws_codeartifact_domain.example.domain,
 *     upstreams: [{
 *         repositoryName: upstream.repository,
 *     }],
 * });
 * ```
 * ### With External Connection
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const upstream = new aws.codeartifact.Repository("upstream", {
 *     repository: "upstream",
 *     domain: aws_codeartifact_domain.test.domain,
 * });
 * const test = new aws.codeartifact.Repository("test", {
 *     repository: "example",
 *     domain: aws_codeartifact_domain.example.domain,
 *     externalConnections: {
 *         externalConnectionName: "public:npmjs",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * CodeArtifact Repository can be imported using the CodeArtifact Repository ARN, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:codeartifact/repository:Repository example arn:aws:codeartifact:us-west-2:012345678912:repository/tf-acc-test-6968272603913957763/tf-acc-test-6968272603913957763
 * ```
 */
class Repository extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["administratorAccount"] = state ? state.administratorAccount : undefined;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["domain"] = state ? state.domain : undefined;
            resourceInputs["domainOwner"] = state ? state.domainOwner : undefined;
            resourceInputs["externalConnections"] = state ? state.externalConnections : undefined;
            resourceInputs["repository"] = state ? state.repository : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
            resourceInputs["upstreams"] = state ? state.upstreams : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.domain === undefined) && !opts.urn) {
                throw new Error("Missing required property 'domain'");
            }
            if ((!args || args.repository === undefined) && !opts.urn) {
                throw new Error("Missing required property 'repository'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["domain"] = args ? args.domain : undefined;
            resourceInputs["domainOwner"] = args ? args.domainOwner : undefined;
            resourceInputs["externalConnections"] = args ? args.externalConnections : undefined;
            resourceInputs["repository"] = args ? args.repository : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["upstreams"] = args ? args.upstreams : undefined;
            resourceInputs["administratorAccount"] = undefined /*out*/;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Repository.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Repository resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Repository(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Repository.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Repository.__pulumiType;
    }
}
exports.Repository = Repository;
/** @internal */
Repository.__pulumiType = 'aws:codeartifact/repository:Repository';
