// @pulumi/aws/location/placeIndex.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlaceIndex = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a Location Service Place Index.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.location.PlaceIndex("example", {
 *     dataSource: "Here",
 *     indexName: "example",
 * });
 * ```
 *
 * ## Import
 *
 * `aws_location_place_index` resources can be imported using the place index name, e.g.
 *
 * ```sh
 *  $ pulumi import aws:location/placeIndex:PlaceIndex example example
 * ```
 */
class PlaceIndex extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state ? state.createTime : undefined;
            resourceInputs["dataSource"] = state ? state.dataSource : undefined;
            resourceInputs["dataSourceConfiguration"] = state ? state.dataSourceConfiguration : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["indexArn"] = state ? state.indexArn : undefined;
            resourceInputs["indexName"] = state ? state.indexName : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
            resourceInputs["updateTime"] = state ? state.updateTime : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.dataSource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataSource'");
            }
            if ((!args || args.indexName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'indexName'");
            }
            resourceInputs["dataSource"] = args ? args.dataSource : undefined;
            resourceInputs["dataSourceConfiguration"] = args ? args.dataSourceConfiguration : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["indexName"] = args ? args.indexName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["tagsAll"] = args ? args.tagsAll : undefined;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["indexArn"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PlaceIndex.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing PlaceIndex resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new PlaceIndex(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PlaceIndex.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PlaceIndex.__pulumiType;
    }
}
exports.PlaceIndex = PlaceIndex;
/** @internal */
PlaceIndex.__pulumiType = 'aws:location/placeIndex:PlaceIndex';
