// @pulumi/aws/qldb/stream.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Stream = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an AWS Quantum Ledger Database (QLDB) Stream resource
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.qldb.Stream("example", {
 *     inclusiveStartTime: "2021-01-01T00:00:00Z",
 *     kinesisConfiguration: {
 *         aggegationEnabled: false,
 *         streamArn: "arn:aws:kinesis:us-east-1:xxxxxxxxxxxx:stream/example-kinesis-stream",
 *     },
 *     ledgerName: "existing-ledger-name",
 *     roleArn: "sample-role-arn",
 *     streamName: "sample-ledger-stream",
 *     tags: {
 *         example: "tag",
 *     },
 * });
 * ```
 */
class Stream extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state ? state.arn : undefined;
            resourceInputs["exclusiveEndTime"] = state ? state.exclusiveEndTime : undefined;
            resourceInputs["inclusiveStartTime"] = state ? state.inclusiveStartTime : undefined;
            resourceInputs["kinesisConfiguration"] = state ? state.kinesisConfiguration : undefined;
            resourceInputs["ledgerName"] = state ? state.ledgerName : undefined;
            resourceInputs["roleArn"] = state ? state.roleArn : undefined;
            resourceInputs["streamName"] = state ? state.streamName : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tagsAll"] = state ? state.tagsAll : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.inclusiveStartTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'inclusiveStartTime'");
            }
            if ((!args || args.kinesisConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kinesisConfiguration'");
            }
            if ((!args || args.ledgerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ledgerName'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            if ((!args || args.streamName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'streamName'");
            }
            resourceInputs["exclusiveEndTime"] = args ? args.exclusiveEndTime : undefined;
            resourceInputs["inclusiveStartTime"] = args ? args.inclusiveStartTime : undefined;
            resourceInputs["kinesisConfiguration"] = args ? args.kinesisConfiguration : undefined;
            resourceInputs["ledgerName"] = args ? args.ledgerName : undefined;
            resourceInputs["roleArn"] = args ? args.roleArn : undefined;
            resourceInputs["streamName"] = args ? args.streamName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["tagsAll"] = args ? args.tagsAll : undefined;
            resourceInputs["arn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Stream.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Stream resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Stream(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Stream.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Stream.__pulumiType;
    }
}
exports.Stream = Stream;
/** @internal */
Stream.__pulumiType = 'aws:qldb/stream:Stream';
