// @pulumi/aws/servicecatalog/tagOptionResourceAssociation.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagOptionResourceAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Service Catalog Tag Option Resource Association.
 *
 * > **Tip:** A "resource" is either a Service Catalog portfolio or product.
 *
 * ## Example Usage
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.servicecatalog.TagOptionResourceAssociation("example", {
 *     resourceId: "prod-dnigbtea24ste",
 *     tagOptionId: "tag-pjtvyakdlyo3m",
 * });
 * ```
 *
 * ## Import
 *
 * `aws_servicecatalog_tag_option_resource_association` can be imported using the tag option ID and resource ID, e.g.,
 *
 * ```sh
 *  $ pulumi import aws:servicecatalog/tagOptionResourceAssociation:TagOptionResourceAssociation example tag-pjtvyakdlyo3m:prod-dnigbtea24ste
 * ```
 */
class TagOptionResourceAssociation extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["resourceArn"] = state ? state.resourceArn : undefined;
            resourceInputs["resourceCreatedTime"] = state ? state.resourceCreatedTime : undefined;
            resourceInputs["resourceDescription"] = state ? state.resourceDescription : undefined;
            resourceInputs["resourceId"] = state ? state.resourceId : undefined;
            resourceInputs["resourceName"] = state ? state.resourceName : undefined;
            resourceInputs["tagOptionId"] = state ? state.tagOptionId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.resourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceId'");
            }
            if ((!args || args.tagOptionId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tagOptionId'");
            }
            resourceInputs["resourceId"] = args ? args.resourceId : undefined;
            resourceInputs["tagOptionId"] = args ? args.tagOptionId : undefined;
            resourceInputs["resourceArn"] = undefined /*out*/;
            resourceInputs["resourceCreatedTime"] = undefined /*out*/;
            resourceInputs["resourceDescription"] = undefined /*out*/;
            resourceInputs["resourceName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TagOptionResourceAssociation.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing TagOptionResourceAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TagOptionResourceAssociation(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TagOptionResourceAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TagOptionResourceAssociation.__pulumiType;
    }
}
exports.TagOptionResourceAssociation = TagOptionResourceAssociation;
/** @internal */
TagOptionResourceAssociation.__pulumiType = 'aws:servicecatalog/tagOptionResourceAssociation:TagOptionResourceAssociation';
