// @pulumi/aws/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.detective = exports.dax = exports.datasync = exports.datapipeline = exports.dataexchange = exports.cur = exports.costexplorer = exports.connect = exports.config = exports.cognito = exports.codestarnotifications = exports.codestarconnections = exports.codepipeline = exports.codedeploy = exports.codecommit = exports.codebuild = exports.codeartifact = exports.cloudwatch = exports.cloudtrail = exports.cloudsearch = exports.cloudhsmv2 = exports.cloudfront = exports.cloudformation = exports.cloudcontrol = exports.cloud9 = exports.chime = exports.cfg = exports.budgets = exports.batch = exports.backup = exports.autoscalingplans = exports.autoscaling = exports.athena = exports.appsync = exports.appstream = exports.apprunner = exports.appmesh = exports.applicationloadbalancing = exports.appflow = exports.appconfig = exports.appautoscaling = exports.apigatewayv2 = exports.apigateway = exports.amplify = exports.amp = exports.alb = exports.acmpca = exports.acm = exports.account = exports.accessanalyzer = void 0;
exports.location = exports.lightsail = exports.licensemanager = exports.lex = exports.lb = exports.lambda = exports.lakeformation = exports.kms = exports.kinesisanalyticsv2 = exports.kinesis = exports.keyspaces = exports.iot = exports.inspector = exports.imagebuilder = exports.identitystore = exports.iam = exports.guardduty = exports.grafana = exports.glue = exports.globalaccelerator = exports.glacier = exports.gamelift = exports.fsx = exports.fms = exports.emrserverless = exports.emrcontainers = exports.emr = exports.elb = exports.elastictranscoder = exports.elasticsearch = exports.elasticloadbalancingv2 = exports.elasticloadbalancing = exports.elasticbeanstalk = exports.elasticache = exports.eks = exports.efs = exports.ecs = exports.ecrpublic = exports.ecr = exports.ec2transitgateway = exports.ec2clientvpn = exports.ec2 = exports.ebs = exports.dynamodb = exports.docdb = exports.dms = exports.dlm = exports.directoryservice = exports.directconnect = exports.devicefarm = void 0;
exports.sqs = exports.sns = exports.simpledb = exports.signer = exports.shield = exports.sfn = exports.ses = exports.servicequotas = exports.servicediscovery = exports.servicecatalog = exports.serverlessrepository = exports.serverless = exports.securityhub = exports.secretsmanager = exports.schemas = exports.sagemaker = exports.s3outposts = exports.s3control = exports.s3 = exports.route53recoveryreadiness = exports.route53recoverycontrol = exports.route53domains = exports.route53 = exports.resourcegroupstaggingapi = exports.resourcegroups = exports.redshiftdata = exports.redshift = exports.rds = exports.ram = exports.quicksight = exports.qldb = exports.pricing = exports.pinpoint = exports.outposts = exports.organizations = exports.opsworks = exports.opensearch = exports.networkmanager = exports.networkfirewall = exports.neptune = exports.mwaa = exports.mskconnect = exports.msk = exports.mq = exports.memorydb = exports.mediastore = exports.mediapackage = exports.mediaconvert = exports.macie2 = exports.macie = void 0;
exports.xray = exports.workspaces = exports.worklink = exports.wafv2 = exports.wafregional = exports.waf = exports.types = exports.transfer = exports.timestreamwrite = exports.synthetics = exports.swf = exports.storagegateway = exports.ssoadmin = exports.ssm = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
// Export members:
__exportStar(require("./arn"), exports);
__exportStar(require("./awsMixins"), exports);
__exportStar(require("./getAmi"), exports);
__exportStar(require("./getAmiIds"), exports);
__exportStar(require("./getArn"), exports);
__exportStar(require("./getAutoscalingGroups"), exports);
__exportStar(require("./getAvailabilityZone"), exports);
__exportStar(require("./getAvailabilityZones"), exports);
__exportStar(require("./getBillingServiceAccount"), exports);
__exportStar(require("./getCallerIdentity"), exports);
__exportStar(require("./getCanonicalUserId"), exports);
__exportStar(require("./getDefaultTags"), exports);
__exportStar(require("./getElasticIp"), exports);
__exportStar(require("./getIpRanges"), exports);
__exportStar(require("./getPartition"), exports);
__exportStar(require("./getPrefixList"), exports);
__exportStar(require("./getRegion"), exports);
__exportStar(require("./getRegions"), exports);
__exportStar(require("./getService"), exports);
__exportStar(require("./provider"), exports);
__exportStar(require("./region"), exports);
__exportStar(require("./tags"), exports);
__exportStar(require("./utils"), exports);
// Export enums:
__exportStar(require("./types/enums"), exports);
// Export sub-modules:
const accessanalyzer = require("./accessanalyzer");
exports.accessanalyzer = accessanalyzer;
const account = require("./account");
exports.account = account;
const acm = require("./acm");
exports.acm = acm;
const acmpca = require("./acmpca");
exports.acmpca = acmpca;
const alb = require("./alb");
exports.alb = alb;
const amp = require("./amp");
exports.amp = amp;
const amplify = require("./amplify");
exports.amplify = amplify;
const apigateway = require("./apigateway");
exports.apigateway = apigateway;
const apigatewayv2 = require("./apigatewayv2");
exports.apigatewayv2 = apigatewayv2;
const appautoscaling = require("./appautoscaling");
exports.appautoscaling = appautoscaling;
const appconfig = require("./appconfig");
exports.appconfig = appconfig;
const appflow = require("./appflow");
exports.appflow = appflow;
const applicationloadbalancing = require("./applicationloadbalancing");
exports.applicationloadbalancing = applicationloadbalancing;
const appmesh = require("./appmesh");
exports.appmesh = appmesh;
const apprunner = require("./apprunner");
exports.apprunner = apprunner;
const appstream = require("./appstream");
exports.appstream = appstream;
const appsync = require("./appsync");
exports.appsync = appsync;
const athena = require("./athena");
exports.athena = athena;
const autoscaling = require("./autoscaling");
exports.autoscaling = autoscaling;
const autoscalingplans = require("./autoscalingplans");
exports.autoscalingplans = autoscalingplans;
const backup = require("./backup");
exports.backup = backup;
const batch = require("./batch");
exports.batch = batch;
const budgets = require("./budgets");
exports.budgets = budgets;
const cfg = require("./cfg");
exports.cfg = cfg;
const chime = require("./chime");
exports.chime = chime;
const cloud9 = require("./cloud9");
exports.cloud9 = cloud9;
const cloudcontrol = require("./cloudcontrol");
exports.cloudcontrol = cloudcontrol;
const cloudformation = require("./cloudformation");
exports.cloudformation = cloudformation;
const cloudfront = require("./cloudfront");
exports.cloudfront = cloudfront;
const cloudhsmv2 = require("./cloudhsmv2");
exports.cloudhsmv2 = cloudhsmv2;
const cloudsearch = require("./cloudsearch");
exports.cloudsearch = cloudsearch;
const cloudtrail = require("./cloudtrail");
exports.cloudtrail = cloudtrail;
const cloudwatch = require("./cloudwatch");
exports.cloudwatch = cloudwatch;
const codeartifact = require("./codeartifact");
exports.codeartifact = codeartifact;
const codebuild = require("./codebuild");
exports.codebuild = codebuild;
const codecommit = require("./codecommit");
exports.codecommit = codecommit;
const codedeploy = require("./codedeploy");
exports.codedeploy = codedeploy;
const codepipeline = require("./codepipeline");
exports.codepipeline = codepipeline;
const codestarconnections = require("./codestarconnections");
exports.codestarconnections = codestarconnections;
const codestarnotifications = require("./codestarnotifications");
exports.codestarnotifications = codestarnotifications;
const cognito = require("./cognito");
exports.cognito = cognito;
const config = require("./config");
exports.config = config;
const connect = require("./connect");
exports.connect = connect;
const costexplorer = require("./costexplorer");
exports.costexplorer = costexplorer;
const cur = require("./cur");
exports.cur = cur;
const dataexchange = require("./dataexchange");
exports.dataexchange = dataexchange;
const datapipeline = require("./datapipeline");
exports.datapipeline = datapipeline;
const datasync = require("./datasync");
exports.datasync = datasync;
const dax = require("./dax");
exports.dax = dax;
const detective = require("./detective");
exports.detective = detective;
const devicefarm = require("./devicefarm");
exports.devicefarm = devicefarm;
const directconnect = require("./directconnect");
exports.directconnect = directconnect;
const directoryservice = require("./directoryservice");
exports.directoryservice = directoryservice;
const dlm = require("./dlm");
exports.dlm = dlm;
const dms = require("./dms");
exports.dms = dms;
const docdb = require("./docdb");
exports.docdb = docdb;
const dynamodb = require("./dynamodb");
exports.dynamodb = dynamodb;
const ebs = require("./ebs");
exports.ebs = ebs;
const ec2 = require("./ec2");
exports.ec2 = ec2;
const ec2clientvpn = require("./ec2clientvpn");
exports.ec2clientvpn = ec2clientvpn;
const ec2transitgateway = require("./ec2transitgateway");
exports.ec2transitgateway = ec2transitgateway;
const ecr = require("./ecr");
exports.ecr = ecr;
const ecrpublic = require("./ecrpublic");
exports.ecrpublic = ecrpublic;
const ecs = require("./ecs");
exports.ecs = ecs;
const efs = require("./efs");
exports.efs = efs;
const eks = require("./eks");
exports.eks = eks;
const elasticache = require("./elasticache");
exports.elasticache = elasticache;
const elasticbeanstalk = require("./elasticbeanstalk");
exports.elasticbeanstalk = elasticbeanstalk;
const elasticloadbalancing = require("./elasticloadbalancing");
exports.elasticloadbalancing = elasticloadbalancing;
const elasticloadbalancingv2 = require("./elasticloadbalancingv2");
exports.elasticloadbalancingv2 = elasticloadbalancingv2;
const elasticsearch = require("./elasticsearch");
exports.elasticsearch = elasticsearch;
const elastictranscoder = require("./elastictranscoder");
exports.elastictranscoder = elastictranscoder;
const elb = require("./elb");
exports.elb = elb;
const emr = require("./emr");
exports.emr = emr;
const emrcontainers = require("./emrcontainers");
exports.emrcontainers = emrcontainers;
const emrserverless = require("./emrserverless");
exports.emrserverless = emrserverless;
const fms = require("./fms");
exports.fms = fms;
const fsx = require("./fsx");
exports.fsx = fsx;
const gamelift = require("./gamelift");
exports.gamelift = gamelift;
const glacier = require("./glacier");
exports.glacier = glacier;
const globalaccelerator = require("./globalaccelerator");
exports.globalaccelerator = globalaccelerator;
const glue = require("./glue");
exports.glue = glue;
const grafana = require("./grafana");
exports.grafana = grafana;
const guardduty = require("./guardduty");
exports.guardduty = guardduty;
const iam = require("./iam");
exports.iam = iam;
const identitystore = require("./identitystore");
exports.identitystore = identitystore;
const imagebuilder = require("./imagebuilder");
exports.imagebuilder = imagebuilder;
const inspector = require("./inspector");
exports.inspector = inspector;
const iot = require("./iot");
exports.iot = iot;
const keyspaces = require("./keyspaces");
exports.keyspaces = keyspaces;
const kinesis = require("./kinesis");
exports.kinesis = kinesis;
const kinesisanalyticsv2 = require("./kinesisanalyticsv2");
exports.kinesisanalyticsv2 = kinesisanalyticsv2;
const kms = require("./kms");
exports.kms = kms;
const lakeformation = require("./lakeformation");
exports.lakeformation = lakeformation;
const lambda = require("./lambda");
exports.lambda = lambda;
const lb = require("./lb");
exports.lb = lb;
const lex = require("./lex");
exports.lex = lex;
const licensemanager = require("./licensemanager");
exports.licensemanager = licensemanager;
const lightsail = require("./lightsail");
exports.lightsail = lightsail;
const location = require("./location");
exports.location = location;
const macie = require("./macie");
exports.macie = macie;
const macie2 = require("./macie2");
exports.macie2 = macie2;
const mediaconvert = require("./mediaconvert");
exports.mediaconvert = mediaconvert;
const mediapackage = require("./mediapackage");
exports.mediapackage = mediapackage;
const mediastore = require("./mediastore");
exports.mediastore = mediastore;
const memorydb = require("./memorydb");
exports.memorydb = memorydb;
const mq = require("./mq");
exports.mq = mq;
const msk = require("./msk");
exports.msk = msk;
const mskconnect = require("./mskconnect");
exports.mskconnect = mskconnect;
const mwaa = require("./mwaa");
exports.mwaa = mwaa;
const neptune = require("./neptune");
exports.neptune = neptune;
const networkfirewall = require("./networkfirewall");
exports.networkfirewall = networkfirewall;
const networkmanager = require("./networkmanager");
exports.networkmanager = networkmanager;
const opensearch = require("./opensearch");
exports.opensearch = opensearch;
const opsworks = require("./opsworks");
exports.opsworks = opsworks;
const organizations = require("./organizations");
exports.organizations = organizations;
const outposts = require("./outposts");
exports.outposts = outposts;
const pinpoint = require("./pinpoint");
exports.pinpoint = pinpoint;
const pricing = require("./pricing");
exports.pricing = pricing;
const qldb = require("./qldb");
exports.qldb = qldb;
const quicksight = require("./quicksight");
exports.quicksight = quicksight;
const ram = require("./ram");
exports.ram = ram;
const rds = require("./rds");
exports.rds = rds;
const redshift = require("./redshift");
exports.redshift = redshift;
const redshiftdata = require("./redshiftdata");
exports.redshiftdata = redshiftdata;
const resourcegroups = require("./resourcegroups");
exports.resourcegroups = resourcegroups;
const resourcegroupstaggingapi = require("./resourcegroupstaggingapi");
exports.resourcegroupstaggingapi = resourcegroupstaggingapi;
const route53 = require("./route53");
exports.route53 = route53;
const route53domains = require("./route53domains");
exports.route53domains = route53domains;
const route53recoverycontrol = require("./route53recoverycontrol");
exports.route53recoverycontrol = route53recoverycontrol;
const route53recoveryreadiness = require("./route53recoveryreadiness");
exports.route53recoveryreadiness = route53recoveryreadiness;
const s3 = require("./s3");
exports.s3 = s3;
const s3control = require("./s3control");
exports.s3control = s3control;
const s3outposts = require("./s3outposts");
exports.s3outposts = s3outposts;
const sagemaker = require("./sagemaker");
exports.sagemaker = sagemaker;
const schemas = require("./schemas");
exports.schemas = schemas;
const secretsmanager = require("./secretsmanager");
exports.secretsmanager = secretsmanager;
const securityhub = require("./securityhub");
exports.securityhub = securityhub;
const serverless = require("./serverless");
exports.serverless = serverless;
const serverlessrepository = require("./serverlessrepository");
exports.serverlessrepository = serverlessrepository;
const servicecatalog = require("./servicecatalog");
exports.servicecatalog = servicecatalog;
const servicediscovery = require("./servicediscovery");
exports.servicediscovery = servicediscovery;
const servicequotas = require("./servicequotas");
exports.servicequotas = servicequotas;
const ses = require("./ses");
exports.ses = ses;
const sfn = require("./sfn");
exports.sfn = sfn;
const shield = require("./shield");
exports.shield = shield;
const signer = require("./signer");
exports.signer = signer;
const simpledb = require("./simpledb");
exports.simpledb = simpledb;
const sns = require("./sns");
exports.sns = sns;
const sqs = require("./sqs");
exports.sqs = sqs;
const ssm = require("./ssm");
exports.ssm = ssm;
const ssoadmin = require("./ssoadmin");
exports.ssoadmin = ssoadmin;
const storagegateway = require("./storagegateway");
exports.storagegateway = storagegateway;
const swf = require("./swf");
exports.swf = swf;
const synthetics = require("./synthetics");
exports.synthetics = synthetics;
const timestreamwrite = require("./timestreamwrite");
exports.timestreamwrite = timestreamwrite;
const transfer = require("./transfer");
exports.transfer = transfer;
const types = require("./types");
exports.types = types;
const waf = require("./waf");
exports.waf = waf;
const wafregional = require("./wafregional");
exports.wafregional = wafregional;
const wafv2 = require("./wafv2");
exports.wafv2 = wafv2;
const worklink = require("./worklink");
exports.worklink = worklink;
const workspaces = require("./workspaces");
exports.workspaces = workspaces;
const xray = require("./xray");
exports.xray = xray;
const provider_1 = require("./provider");
pulumi.runtime.registerResourcePackage("aws", {
    version: utilities.getVersion(),
    constructProvider: (name, type, urn) => {
        if (type !== "pulumi:providers:aws") {
            throw new Error(`unknown provider type ${type}`);
        }
        return new provider_1.Provider(name, undefined, { urn });
    },
});
